### PR TITLE
[nGraph Transformations] NMS convert precisions - change sequence of checks

### PR DESCRIPTION
**Pull request by alvoron**
### Details:
If model contains NMS node and being executed on ARM CPU with f16 inference precision then output precision issues may occur: `The plugin does not support I64` or `While validating node ABC type of then_body output i32 is not equal type of else_body output i64`
The issue happens because NMS node wrapped by `TypeRelaxed` is not overridden output types while i32/i64 conversion.
The suggested fix is to change the order of checks: `TypeRelaxed` check initially, i32/i64 check then and `TypeRelaxed` creation at the end.

### Tickets:
 - 123909
 - 123910

### CODE REVIEW

**Comment by itikhono** [comment]
could you add a test?

**Comment by alvoron** [comment]
> could you add a test?

Added

**Review by itikhono — APPROVED** [review]

File: src/common/transformations/src/transformations/convert_precision.cpp [hunk 1]
```diff
@@ -606,6 +606,19 @@ bool fuse_type_to_nms5(const std::shared_ptr<ov::Node>& node, const precisions_m
     }
 
     bool res = false;
+    auto type_relaxed = std::dynamic_pointer_cast<ov::op::TypeRelaxedBase>(node);
```
Comment (itikhono) [inline src/common/transformations/src/transformations/convert_precision.cpp:609]: do we have other similar implementations in this transformation?
In which it's also worth moving that part up

Comment (alvoron) [reply src/common/transformations/src/transformations/convert_precision.cpp:609]: I haven't found similar implementations in the transformation.

**Pull request merged**